### PR TITLE
Htk spatial mapping refactor

### PR DESCRIPTION
**Pull request by HodgsonSDAS**
1. Refactored Spatial Mapping Scripts into  their own namespace: `HoloToolkit.Unity.SpatialMapping`
2. Removed imported materials from FakeSpatialMappingMesh && SRMesh.
3. Organized project files in test folder.
4. Removed some hidden parameter warnings from https://github.com/Microsoft/HoloToolkit-Unity/issues/210
5. Misc cleanup and formatting in Spatial Mapping related classes.

New Folder Layout:
- SpatialMapping
    - Materials
    - Plugins
    - Prefabs
    - Scripts
        - RemoteMapping
        - SpatialProcessing
    - Shaders
    - Tests
        - Meshes
        - Prefabs
        - Scenes
        - Scripts

### CODE REVIEW

**Comment by keveleigh** [comment]
TapToPlace.cs is something that had been requested by the community. We should keep it in the actual Scripts folder, though I'm certainly not against refactoring it!

**Comment by HodgsonSDAS** [comment]
Okay, I'll revert. But I'd like to open an issue to refactor into something more generic.

I think the way it is it's really not that useful.  I've found myself just writing my own implementations off a base class I made myself.

**Comment by keveleigh** [comment]
For the purpose of making it better, what do you find not useful about it?

**Comment by HodgsonSDAS** [comment]
I'd like to put my own logic into `Start`, `OnInputClicked`, and `Update`.

**Comment by keveleigh** [comment]
Makes sense! We actually adapted it off https://github.com/Microsoft/HolographicAcademy/blob/Holograms-101/Catch%20up%20Chapters/8-Finale/Assets/Scripts/TapToPlaceParent.cs, which does what you mentioned, but converted it into the very simple actions of picking up and placing an object.

File: Assets/HoloToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMeshTarget.cs [hunk 1]
```diff
@@ -103,16 +103,16 @@ void Update()
...
                         }
                     }
 
                     // Finally disconnect.
-                    ClientConnected = false;
+                    clientConnected = false;
                     networkClient.Close();
 
                     // And wait for the next connection.
-                    AsyncCallback callback = new AsyncCallback(OnClientConnect);
```
Comment (keveleigh) [inline Assets/HoloToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMeshTarget.cs]: We don't need the explicit new AsyncCallback here anymore?

File: Assets/HoloToolkit/SpatialMapping/Tests/Support/Materials/lambert1.mat [hunk 2]
```diff
@@ -1,127 +0,0 @@
-%YAML 1.1
```
Comment (keveleigh) [inline Assets/HoloToolkit/SpatialMapping/Tests/Support/Materials/lambert1.mat]: Will this and defaultMat.mat keep regenerating themselves when the fbxs are imported?

Comment (S-Hodgson-MMOARgames) [inline Assets/HoloToolkit/SpatialMapping/Tests/Support/Materials/lambert1.mat]: No I disabled the regeneration of the materials by the fbxs.

File: Assets/HoloToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMeshTarget.cs (hunk 1 above)
Comment (S-Hodgson-MMOARgames) [inline Assets/HoloToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMeshTarget.cs]: The new AsyncCallback is redundant.

File: Assets/HoloToolkit/SpatialMapping/Tests/Support/Materials/lambert1.mat (hunk 2 above)
Comment (keveleigh) [inline Assets/HoloToolkit/SpatialMapping/Tests/Support/Materials/lambert1.mat]: Looks good then!

**Pull request merged**